services/pdf-ms/helpers/pdf.ts

```diff
@@ -50,31 +50,40 @@ const replaceTextInTextObject = (textObject: TextObject, json: Record<string, an
   const templateStrings = [...textObject.text.matchAll(regex)];
   const newTextObject: TextObject = { ...textObject };
 
-  const newText = templateStrings.reduce((prev, currentRegexResult) => {
+  const newText = templateStrings.reduce((previous, currentRegexResult) => {
     const replacement = getPropertyFromDottedString(json, currentRegexResult[1].trim());
     if (replacement === undefined || replacement === 'undefined') {
-      return prev.replace(currentRegexResult[0], '');
+      return previous.replace(currentRegexResult[0], '');
     }
-    return prev.replace(currentRegexResult[0], replacement);
+    return previous.replace(currentRegexResult[0], replacement);
   }, textObject.text);
   newTextObject.text = newText;
   return newTextObject;
 };
 
 /** Takes a base pdf, a templateData object that tells us where to put the texts,
- * and a jsonData object from which we import the values. */
+ * and a json from which we import the values. */
 export const modifyPdf = async (
   pdfBuffer: Buffer,
   template: Template,
-  jsonData: Record<string, any>
+  json: Record<string, any>,
+  changedValues?: string[],
+  newValues?: string[]
 ) => {
   const document = await PDFDocument.load(pdfBuffer);
   const pages = document.getPages();
   const fonts = await loadFonts(document);
 
-  const replacedTextObjects = template.texts.map(textObject =>
-    replaceTextInTextObject(textObject, jsonData)
-  );
+  const replacedTextObjects = template.texts.map(textObject => {
+    const replacedText = replaceTextInTextObject(textObject, json);
+    if (
+      replacedText.valueId &&
+      (newValues.includes(replacedText.valueId) || changedValues.includes(replacedText.valueId))
+    ) {
+      replacedText.color = rgb(174 / 255, 11 / 255, 5 / 255);
```

Comment:
const?

Reply:
the variabel replacedText is an object, and we do only change properties in the object. Therefore using const here is fine for me. If we where to write over the value of replacedText then a let would be the way to go.

Reply:
> the variabel replacedText is an object, and we do only change properties in the object. Therefore using const here is fine for me. If we where to write over the value of replacedText then a let would be the way to go.

I mean, put color in some const variable stored in the global const lib file

Reply:
Ah, okay, thats what you mean. Then I agree, both color values should be lifted to two consts. But not sure it needs to go to some global lib file, since they are probably only used in this one file, so local constants feels better to me.